autodispose/src/test/java/com/uber/autodispose/AutoDisposeParallelFlowableTest.java

```diff
@@ -0,0 +1,365 @@
+package com.uber.autodispose;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+
+import java.util.List;
+
+import io.reactivex.Flowable;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Predicate;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.MaybeSubject;
+import io.reactivex.subscribers.TestSubscriber;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class AutoDisposeParallelFlowableTest {
+
+  private static final int DEFAULT_PARALLELISM = 2;
+
+  @Rule public RxErrorsRule rule = new RxErrorsRule();
+
+  @After public void resetPlugins() {
+    AutoDisposePlugins.reset();
+  }
+
+  @Test public void autoDispose_withMaybe_normal() {
+    TestSubscriber<Integer> firstSubscriber = new TestSubscriber<>();
+    TestSubscriber<Integer> secondSubscriber = new TestSubscriber<>();
+    PublishProcessor<Integer> source = PublishProcessor.create();
+    MaybeSubject<Integer> lifecycle = MaybeSubject.create();
+
+    //noinspection unchecked
+    Subscriber<Integer>[] subscribers = new Subscriber[] {firstSubscriber, secondSubscriber};
+    source
+        .parallel(DEFAULT_PARALLELISM)
+        .as(AutoDispose.<Integer>autoDisposable(lifecycle))
+        .subscribe(subscribers);
+    firstSubscriber.assertSubscribed();
+    secondSubscriber.assertSubscribed();
+
+    assertThat(source.hasSubscribers()).isTrue();
+    assertThat(lifecycle.hasObservers()).isTrue();
+
+    source.onNext(1);
+    source.onNext(2);
+    firstSubscriber.assertValue(1);
+    secondSubscriber.assertValue(2);
```

Comment:
Can you explain this part a bit? Not quite sure I followed how this works (probably my lack of familiarity with `ParallelFlowable`).

Reply:
So the `ParallelFlowable` emits the items to the subscribers in a round robin fashion, and here we are emitting 1 & 2 to two subscribers. This means that the `firstSubscriber` will only receive 1 and the `secondSubscriber` would only receive 2 and nothing else.

Reply:
gotcha :+1: